modular_skyrat/modules/deployables/code/deployable.dm

```diff
@@ -0,0 +1,701 @@
+			stop_mode()
+		if(DRAIN_MODE)
+			balloon_alert(clicker, "draining...")
+			drain_mode()
+		if(PUMP_MODE)
+			balloon_alert(clicker, "pumping...")
+			pump_mode()
+		if(FOLD)
+			balloon_alert(clicker, "folding up...")
+			if(do_after(clicker, 3 SECONDS, src))
+				balloon_alert_to_viewers("resetting...")
+				undeploy()
+
+/obj/structure/bed/borg_action_pacifier/proc/check_menu(mob/user)
+	if(!istype(user))
+		CRASH("A non-mob is trying to issue an order.")
+	if(user.incapacitated() || !can_see(user, src) || user == buckled_cyborg)
+		return FALSE
+	return TRUE
+
+// Functions for the radial choices
+/obj/structure/bed/borg_action_pacifier/process()
+	if(!buckled_cyborg)
+		return
+
+	/// The cyborg's current cell
+	var/obj/item/stock_parts/cell/cell = buckled_cyborg.cell
+
+	switch(enabled_function)
+		if(DRAIN_MODE)
+			if(cell.charge > 0)
+				drain_cell(cell, TRANSFER_INC)
+
+			if(cell.charge <= 0)
+				cell.charge = 0
+				stop_mode()
+
+			else if(power_storage > MAX_POWER)
+				power_storage = MAX_POWER
+				stop_mode()
+
+		if(PUMP_MODE)
+			if(power_storage > 0)
+				pump_cell(cell, TRANSFER_INC)
+
+			if(power_storage <= 0)
+				power_storage = 0
+				stop_mode()
+
+			else if(cell.charge > cell.maxcharge)
+				cell.charge = cell.maxcharge
+				stop_mode()
+
+		if(NONE)
+			return
+
+/obj/structure/bed/borg_action_pacifier/proc/drain_cell(obj/item/stock_parts/cell/cell)
+	if(locked)
+		render_lock()
+	else
+		buckled_cyborg.regenerate_icons()
+
+	cell.charge = cell.charge - TRANSFER_INC
+	power_storage = power_storage + TRANSFER_INC
+
+	playsound(src, 'modular_skyrat/master_files/sound/effects/robot_drain.ogg', 25, FALSE, falloff_exponent = 10)
+	do_sparks(1, TRUE, buckled_cyborg)
+
+/obj/structure/bed/borg_action_pacifier/proc/pump_cell(obj/item/stock_parts/cell/cell)
+	if(locked)
+		render_lock()
+	else
+		buckled_cyborg.regenerate_icons()
+
+	power_storage = power_storage - TRANSFER_INC
+	cell.charge = cell.charge + TRANSFER_INC
+
+	playsound(src, 'modular_skyrat/master_files/sound/effects/robot_pump.ogg', 25, FALSE, falloff_exponent = 10)
+	do_sparks(1, TRUE, buckled_cyborg)
+
+/obj/structure/bed/borg_action_pacifier/proc/drain_mode()
+	enabled_function = DRAIN_MODE
+
+/obj/structure/bed/borg_action_pacifier/proc/pump_mode()
+	enabled_function = PUMP_MODE
+
+/obj/structure/bed/borg_action_pacifier/proc/stop_mode()
+	playsound(src, 'sound/machines/ping.ogg', 50, FALSE, falloff_exponent = 10)
+	enabled_function = NONE
+
+/obj/structure/bed/borg_action_pacifier/proc/lock(mob/living/clicker)
+	if(clicker)
+		log_combat(clicker, buckled_cyborg, "locked down cyborg")
+
+	playsound(src, 'modular_skyrat/master_files/sound/effects/robot_lock.ogg', 50, TRUE, falloff_exponent = 10)
+	buckled_cyborg.SetLockdown(TRUE)
+	locked = TRUE
+	render_lock()
+
+/obj/structure/bed/borg_action_pacifier/proc/unlock(mob/living/clicker)
+	if(clicker)
+		log_combat(clicker, buckled_cyborg, "released cyborg")
+
+	buckled_cyborg.SetLockdown(FALSE)
```

Comment:
this will remove lockdown even if they were externally locked down by the robo console/wire hacking

Reply:
intended, the device is a manual override